File: src/Qwack.Transport/TransportObjects/MarketData/Curves/TO_BasisPriceCurve.cs
using System;
using System.Collections.Generic;
using System.Text;
using ProtoBuf;
using Qwack.Transport.BasicTypes;
using Qwack.Transport.TransportObjects.Instruments;

namespace Qwack.Transport.TransportObjects.MarketData.Curves
{
    [ProtoContract]
    public class TO_BasisPriceCurve
    {
        public TO_PriceCurve BaseCurve { get; set; }
        public TO_PriceCurve Curve { get; set; }
        public DateTime BuildDate { get; set; }
        public string Name { get; set; }
        public string AssetId { get; set; }
        public string Currency { get; set; }
        public PriceCurveType CurveType { get; set; }
        public List<TO_Instrument> Instruments { get; set; }
        public List<DateTime> Pillars { get; set; }
        public List<string> PillarLabels { get; set; }
        public TO_IrCurve DiscountCurve { get; set; }
        public string SpotLag { get; set; }
        public string SpotCalendar { get; set; }
        public CommodityUnits Units { get; set; }

    }
}

File: src/Qwack.Transport/GeneratedTypeScript/TO_BasisPriceCurve.ts

import { TO_PriceCurve } from './TO_PriceCurve';
import { PriceCurveType } from './PriceCurveType';
import { TO_Instrument } from './TO_Instrument';
import { TO_IrCurve } from './TO_IrCurve';
import { CommodityUnits } from './CommodityUnits';

export interface TO_BasisPriceCurve  {
	 baseCurve: TO_PriceCurve;
	 curve: TO_PriceCurve;
	 buildDate: Date;
	 name: string;
	 assetId: string;
	 currency: string;
	 curveType: PriceCurveType;
	 instruments: TO_Instrument[];
	 pillars: Date[];
	 pillarLabels: string[];
	 discountCurve: TO_IrCurve;
	 spotLag: string;
	 spotCalendar: string;
	 units: CommodityUnits;
}


